python/lsst/obs/base/butler_tests.py

```diff
@@ -0,0 +1,186 @@
+from __future__ import absolute_import, division, print_function
+from future.utils import with_metaclass
+#
+# LSST Data Management System
+# Copyright 2016 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import abc
+import inspect
+import unittest
+import collections
+
+
+class ButlerGetTests(with_metaclass(abc.ABCMeta)):
+    """
+    Tests of obs_* Butler get() functionality.
+
+    In the subclasses's setUp():
+        * Instantiate a Butler on the testdata repository as self.butler.
+        * Set self.dataIds to a dictionary of exposure names:dataId of that
+          exposure in the testdata repository, with unittest.SkipTest as the
+          value for any exposures you do not have/do not want to test. It must
+          contain a valid 'raw' dataId, in addition to 'bias','flat','dark',
+          which may be set to SkipTest. For example::
+              self.dataIds = {'raw': {'visit': 1, 'filter': 'g'},
+                              'bias': {'visit': 1},
+                              'flat': {'visit': 1},
+                              'dark': unittest.SkipTest
+        * Call setUp_butlerGet() to fill in required parameters.
+    """
+
+    def setUp_butlerGet(self,
+                        ccdExposureId_bits=None,
+                        exposureIds=None,
+                        filters=None,
+                        exptimes=None,
+                        detectorIds=None,
+                        detector_names=None,
+                        detector_serials=None,
+                        dimensions=None,
+                        sky_origin=None,
+                        raw_subsets=None,
+                        good_ccds=None,
+                        bad_ccds=None,
+                        linearizer=None
+                        ):
+        """
+        Set up the necessary variables for butlerGet tests.
+
+        ccdExposureId_bits: int
+            expected value of ccdExposureId_bits
+        exposureIds: dict
+            dict of exposure: exposure_id
```

Comment:
I've clarified it to be "exposure name", and added a note about it relating to self.dataIds. Also clarified exposure_id.